alert-common/src/main/java/com/synopsys/integration/alert/common/action/api/AbstractConfigResourceActions.java

```diff
@@ -0,0 +1,182 @@
+        this.authorizationManager = authorizationManager;
+    }
+
+    protected abstract ActionResponse<FieldModel> createResource(FieldModel resource);
+
+    protected abstract ActionResponse<FieldModel> deleteResource(Long id);
+
+    protected abstract ActionResponse<List<FieldModel>> readAllResources();
+
+    protected abstract ActionResponse<List<FieldModel>> readAllByContextAndDescriptor(String context, String descriptorName);
+
+    protected abstract Optional<FieldModel> findFieldModel(Long id);
+
+    protected abstract ValidationActionResponse testResource(FieldModel resource);
+
+    protected abstract ActionResponse<FieldModel> updateResource(Long id, FieldModel resource);
+
+    protected abstract ValidationActionResponse validateResource(FieldModel resource);
+
+    public ActionResponse<List<FieldModel>> getAllByContextAndDescriptor(String context, String descriptorName) {
+        if (!authorizationManager.hasReadPermission(context, descriptorName)) {
+            return new ActionResponse<>(HttpStatus.FORBIDDEN, AbstractResourceActions.FORBIDDEN_MESSAGE);
+        }
+        return readAllByContextAndDescriptor(context, descriptorName);
+    }
+
+    @Override
+    public ActionResponse<FieldModel> create(FieldModel resource) {
+        if (!authorizationManager.hasCreatePermission(resource.getContext(), resource.getDescriptorName())) {
+            return new ActionResponse<>(HttpStatus.FORBIDDEN, AbstractResourceActions.FORBIDDEN_MESSAGE);
+        }
+        ValidationActionResponse validationResponse = validateResource(resource);
+        if (validationResponse.isError()) {
+            return new ActionResponse<>(validationResponse.getHttpStatus(), validationResponse.getMessage().orElse(null));
+        }
+        return createResource(resource);
+    }
+
+    @Override
+    public ActionResponse<List<FieldModel>> getAll() {
+        try {
+            Set<String> descriptorNames = descriptorAccessor.getRegisteredDescriptors()
+                                              .stream()
+                                              .map(RegisteredDescriptorModel::getName)
+                                              .collect(Collectors.toSet());
+            if (!authorizationManager.anyReadPermission(List.of(ConfigContextEnum.DISTRIBUTION.name(), ConfigContextEnum.GLOBAL.name()), descriptorNames)) {
+                return new ActionResponse<>(HttpStatus.FORBIDDEN, AbstractResourceActions.FORBIDDEN_MESSAGE);
+            }
+            return readAllResources();
+        } catch (AlertException ex) {
+            return new ActionResponse<>(HttpStatus.INTERNAL_SERVER_ERROR, String.format("Error reading configurations: %s", ex.getMessage()));
+        }
+    }
+
+    @Override
+    public ActionResponse<FieldModel> getOne(Long id) {
+        if (null == id) {
+            return new ActionResponse<>(HttpStatus.BAD_REQUEST, AbstractResourceActions.RESOURCE_IDENTIFIER_MISSING);
+        }
```

Comment:
Based on the way this class is used (i.e. in a controller) is this case actually possible? If the `RequestMapping` requires an id, and an id is missing from the request, then it will either be a call to the "getAll" endpoint (if one exists) or Spring will return a 404.

Reply:
Just tested this out Spring throws an exception if it's the wrong type so I will remove that.